conn.go

```diff
@@ -1324,11 +1323,15 @@ func (c *Conn) awaitSchemaAgreement() (err error) {
 		}
 
 	cont:
-		time.Sleep(200 * time.Millisecond)
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-time.After(200 * time.Millisecond):
+		}
```

Comment:
From https://godoc.org/time#After : "The underlying Timer is not recovered by the garbage collector until the timer fires."

So I would suggest chaining contexts like this here.
e.g. 
```go
	waitForContextOrTimeout := func(ctx context.Context, d time.Duration) error {
		waitingCtx, cancel := context.WithTimeout(d)
		defer cancel()
		<-waitingCtx.Done()
		if err := waitingCtx.Err(); err == ctx.Err() {
			return err
		}
		return nil
	}
	if err := waitForContextOrTimeout(ctx, 200 * time.Millisecond); err != nil {
		return err
	}
}
```

Reply:
200ms wait should be fine and I dont expect this case to be hit frequently, we could replace this with a NewTimer/Stop/Reset setup but I think that complicates it already, is this really a problem?